### DISCUSSION TITLE
How to disable some messages?

### DISCUSSION THREAD

**Discussion by jafc76** [nlef/moonraker-telegram-bot#216] [Q&A]
Hi, I love the the bot but I'm getting too many notifications.
How can I disable some of them?
I don't need this kind of messages for example:

- Printer online

- Klippy changed state to shutdown
Lost communication with MCU 'mcu'
Once the underlying issue is corrected, use the
"FIRMWARE_RESTART" command to reset the firmware, reload the
config, and restart the host software.
Printer is shutdown

- Klippy changed state to ready

- Bot online, no moonraker connection!
 Connection failed. Not Found 
Failing...

Thanks in advance!

**Comment by aka13-404** [comment]
What you list are error messages. You can not disable error messages. 
Since error messages vary, and change, it is highly unlikely that they ever be configurable on a per-message basis.

**Reply by jafc76** [reply]
Thanks for your answer.
Yes, are error messages but it's because I switch off the printer with a TP-LINK smart plug or because I reboot the service for a configuration change so I'm getting many notifications that I don't need. I understand an error notification when it's printing but if I turn off the smart plug or restart a service I don't wanna get messages. I have too many messages and what I really want is to check the status, to use the time lapse and to get a notification if during the printing is any error and when the print is finished.
Would be nice to customize editing a file, what messages the bot is sending.

**Reply by Paradisebaker** [reply]
I second this request.
especially when doing a lot of optimization work on the printer, you are constantly restarting klipper, rebooting the pi, messing with your printer, etc., all resulting in a flood of messages coming.
And even error messages could be filtered by checking the bool on a variable in a file, (e.g.: if "notification_startup=0" (or similar) then dont send a notification) and then either sending the message or not.
In the Octoprint plugin, there is the possibility to select the events that should trigger notifications, as well as the possible commands. this could be something similar.
The MCU error could also be limited to only trigger a notification if a print is currently running. 
If a config file would be too much trouble for this, could you maybe point out the section of the related .py file that one could comment out? At own risk of functionality of course!

Thank you very much.

**Comment by Paradisebaker** [comment]
I did find a way to get rid of the most annoying ones, I commented out a section in this python script, which I suspect to take care of the error messages:
/home/<klipper-user>/moonraker-telegram-bot/bot/websocket_helper.py

in the section 
_def websocket_to_message(self, ws_loc, ws_message):_

I left the status part alone, but did comment this part out (sorry for the strange formatting, but I seem not to be able to get everything into the code format):
`
        if "id" in json_message:
            if "result" in json_message:
                message_result = json_message["result"]

                if "status" in message_result:
                    self.status_response(message_result["status"])
                    return

                #if "state" in message_result:
                #    klippy_state = message_result["state"]
                #    self._klippy.state = klippy_state
                #    if klippy_state == "ready":
                #        if ws_loc.keep_running:
                #            self._klippy.connected = True
                #            if self._klippy.state_message:
                #                self._notifier.send_error(f"Klippy changed state to {self._klippy.state}")
                #                self._klippy.state_message = ""
                #            self.subscribe(ws_loc)
                #            if self._scheduler.get_job("ws_reschedule"):
                #                self._scheduler.remove_job("ws_reschedule")
                #    elif klippy_state in ["error", "shutdown", "startup"]:
                #        self._klippy.connected = False
                #        self._scheduler.add_job(
                #            self.reshedule,
                #            "interval",
                #            seconds=2,
                #            id="ws_reschedule",
                #            replace_existing=True,
                #        )
                #        state_message = message_result["state_message"]
                #        if self._klippy.state_message != state_message and klippy_state != "startup":
                #            self._klippy.state_message = state_message
                #            self._notifier.send_error(f"Klippy changed state to {self._klippy.state}\n{self._klippy.state_message}")
                #    else:
                #        logger.error("UnKnown klippy state: %s", klippy_state)
                #        self._klippy.connected = False
                #        self._scheduler.add_job(
                #            self.reshedule,
                #            "interval",
                #            seconds=2,
                #            id="ws_reschedule",
                #            replace_existing=True,
                #        )
                #    return

                if "devices" in message_result:
                    for device in message_result["devices"]:
                        self.power_device_state(device)
                    return

            if "error" in json_message:
                self._notifier.send_error(f"{json_message['error']['message']}")
`

Anyhow, this will then also not send a notification, if a real error occurs. if someone could maybe adapt the script to do some check if a print job is running and only then send error notifications, that would be great. my coding skills are by far not good enough to do that.
(happy that I managed to get this far....)
